clarity.d.ts

```diff
@@ -73,12 +73,16 @@ interface IPayload {
 }
 
 interface IEnvelope {
+  impressionId: string;
+  time?: number;
+  sequenceNumber?: number;
+}
+
+interface IClientInfo {
   clarityId: string;
   impressionId: string;
   url: string;
   version: string;
-  time?: number;
-  sequenceNumber?: number;
```

Comment:
Why are these nullable? I realize that is before this change but curious why.

Reply:
All other properties in the envelope are constant throughout the impression, so we were storing an "abstract" envelope object without time and sequenceNo, which was later used in various places by appending relevant time and sequenceNo to it. 

This behavior is optimized now to avoid this pattern.

---
In reply to: [144390698](https://github.com/Microsoft/clarity-js/pull/39#discussion_r144390698) [](ancestors = 144390698)